File: cases/taint/java/model-declared-source-negative/ModelDeclaredSourceNegative.java
package dataflowbench.taint;

final class Config {
    static String fetchRemote() {
        return "r";
    }

    static String fetchLocal() {  // DFB-SOURCE: model-declared-source-input
        return "l";
    }
}

final class ModelDeclaredSourceNegative {
    static void dfb_sink(String value) { }  // DFB-SINK: model-declared-source-sink

    static void run() {
        dfb_sink(Config.fetchLocal());
    }
}

File: cases/taint/javascript/model-declared-source-negative/ModelDeclaredSourceNegative.js
const Config = {
  fetchRemote: function fetchRemote() {
    return "r";
  },
  fetchLocal: function fetchLocal() { // DFB-SOURCE: model-declared-source-input
    return "l";
  }
};

function dfb_sink(value) {} // DFB-SINK: model-declared-source-sink

function run() {
  dfb_sink(Config.fetchLocal());
}
